crates/red_knot_project/src/metadata.rs

```diff
@@ -551,6 +589,176 @@ expected `.`, `]`
 
         Ok(())
     }
+    #[test]
+    fn requires_python_no_python_version() -> anyhow::Result<()> {
+        let system = TestSystem::default();
+        let root = SystemPathBuf::from("/app");
+
+        system
+            .memory_file_system()
+            .write_file(
+                root.join("pyproject.toml"),
+                r#"
+                [project]
+                requires-python = ">=3.12"
+                "#,
+            )
+            .context("Failed to write file")?;
+
+        let root = ProjectMetadata::discover(&root, &system)?;
+
+        snapshot_project!(root);
```

Comment:
Minor: do these tests need to be snapshot tests? I would find them more readable if they would include a simple `assert_eq!`, I think. Something like
```suggestion
assert_eq!(
            root.options
                .environment
                .unwrap()
                .python_version
                .unwrap()
                .to_string(),
            "3.12"
        );
```